### DISCUSSION TITLE
Y-Axis labels not falling to negative values.

### DISCUSSION THREAD

**Discussion by binshadpix** [patrykandpatrick/vico#979] [Questions]
### Question

Providing y values as
 axisValuesOverrider = AxisValuesOverrider.fixed(
                    minY = -1f,
                    maxY = 5f
                )
                
but the line chart Y-Axis values are starting from 0
![Screenshot 2024-12-26 at 3 51 53 PM](https://github.com/user-attachments/assets/b900164e-c996-4443-b385-f7719f18bb48)

### Vico version(s)

1.12.0

### UI framework(s)

Jetpack Compose

**Comment by binshadpix** [comment]
The itemPlacer count is 4

Expected behaviour for yAxis is

5
3
1
-1

**Comment by Gowsky** [comment]
Hello, @binshadpix. The default `AxisItemPlacer.Vertical` implementation treats the positive and negative parts separately to ensure that _y_ = 0 is labeled, which is usually desired. As you can see in the screenshot, with this label present, there's insufficient room left at the bottom, so the three remaining labels are all added at the top.

To achieve the behavior you've described, without _y_ = 0 labeled, you can create a simple custom `AxisItemPlacer.Vertical` implementation. Feel free to also create a feature request for such an implementation to be built into the library, but please note that new functionality is added only to Vico 2. Vico 1 receives only bug fixes.

**Reply by binshadpix** [reply]
Hai @Gowsky Thanks for the reply
Can you please guide me to create a custom AxisItemPlacer.Vertical for this case.

And, FYI i have updated vico library to 2.0.0-beta.3(The mentioned version in the question was 1.12.0)

**Reply by Gowsky** [reply]
In version 2 it's `VerticalAxis.ItemPlacer`. All of its functions are documented. Please see [here](https://www.patrykandpatrick.com/vico/api/latest/core/com.patrykandpatrick.vico.core.cartesian.axis/-vertical-axis/-item-placer/index.html). If you have specific questions, let me know.

**Reply by binshadpix** [reply]
Hi @Gowsky still i didnot get a solution to omit the unnecessary 0 in yaxis

**Reply by Gowsky** [reply]
It's mentioned above. Please see [`VerticalAxis.ItemPlacer.getLabelValues`](https://www.patrykandpatrick.com/vico/api/latest/core/com.patrykandpatrick.vico.core.cartesian.axis/-vertical-axis/-item-placer/index.html#639902797%2FFunctions%2F186187275). If you need further help, please ask specific questions.

**Reply by binshadpix** [reply]
Thanks @Gowsky Got it!!!

**Reply by Gowsky** [reply]
You're welcome!
